lib/goth/server.ex

```diff
@@ -57,18 +50,49 @@ defmodule Goth.Server do
       |> Map.replace!(:backoff, Backoff.new(backoff_opts))
       |> Map.replace!(:retries, state.max_retries)
 
+    case prefetch || :sync do
+      :async ->
+        {:ok, state, {:continue, :async_prefetch}}
+
+      :sync ->
+        do_prefetch(state)
+        {:ok, state}
+    end
+  end
+
+  @impl true
+  def handle_continue(:async_prefetch, state) do
+    do_prefetch(state)
+    {:noreply, state}
+  end
+
+  defp do_prefetch(state) do
```

Comment:
let's skip the `do_`:

```suggestion
  defp prefetch(state) do
```